### PR TITLE
header_rewrite: allow for use of maxminddb as source of geo truth

### PR DESCRIPTION
**Pull request by randall**

### CODE REVIEW

File: plugins/healthchecks/Makefile.inc [hunk 1]
```diff
@@ -14,9 +14,5 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 
-if BUILD_HEALTHCHECK_PLUGIN
```
Comment (randall) [inline plugins/healthchecks/Makefile.inc:17]: unrelated, but general cleanup. we conditionalize compilation at the parent Makefile for everything else.

**Comment by randall** [comment]
[approve ci]

**Comment by randall** [comment]
[approve ci]

**Comment by randall** [comment]
[approve ci]

**Comment by randall** [comment]
[approve ci]

**Comment by randall** [comment]
[approve ci]

**Comment by randall** [comment]
[approve ci autest]
[approve ci clang-analyzer]

**Comment by SolidWallOfCode** [comment]
Need to have a way to not use the library even if it's installed.

**Comment by bryancall** [comment]
@ezelkow1 do you have experience with maxmind?

File: plugins/header_rewrite/header_rewrite.cc [hunk 2]
```diff
@@ -329,16 +319,31 @@ TSPluginInit(int argc, const char *argv[])
...
+    }
   }
 
   // Parse the global config file(s). All rules are just appended
   // to the "global" Rules configuration.
   RulesConfig *conf = new RulesConfig;
   bool got_config   = false;
 
-  initGeoIP();
+  std::call_once(initGeoLibs, [&dbPath]() { initGeoLib(dbPath); });
```
Comment (SolidWallOfCode) [inline plugins/header_rewrite/header_rewrite.cc:344]: It's OK to call this from global init and remap init?

**Comment by ezelkow1** [comment]
> 
> 
> @ezelkow1 do you have experience with maxmind?

yea, but its mostly similar to what was done here, which is similar to what I did in the maxmind_acl plugin. I briefly looked through the PR and it looked good to me, though Im not sure what @SolidWallOfCode comment meant. Do you mean we need a way to have a --disable-maxmind type of config to be able to not bring it in if it exists?

**Comment by ezelkow1** [comment]
Also while you are in there, it may be good to add some notes in the docs re: https://github.com/apache/trafficserver/issues/7669

Since both geoip and mmdb are using mmap, if you swap the db out from underneath the plugin while it is running you may end up with a bad time since its using an mmap'd version to the file location @pbchou found

**Comment by randall** [comment]
> Need to have a way to not use the library even if it's installed.

@SolidWallOfCode Can you provide more detail? do you mean at configure time? eg `--disable-maxmind` or at runtime?

**Comment by randall** [comment]
Latest version will return `unknown` and `-1` if `--geo-db-path` is not specified on command line, resulting in behavior like the current maxmind implementation(for which there is no switch to disable currently)

**Comment by randall** [comment]
> Also while you are in there, it may be good to add some notes in the docs re: #7669
> 

I guess I don't see an issue with the current implementation (overwriting data files with `cp` is usually a bad idea and not unique to this plugin or program). If you use a package manager(like rpm) or something like saltstack to manage your db packages , they both do an atomic update of the file (renaming the new file into place)

**Review by ezelkow1 — APPROVED** [review]

**Comment by zwoop** [comment]
Cherry-picked to v9.1.x branch.

**Pull request merged**